tests/fe/fe_values_view_25b_nonsymmetric.cc

```diff
@@ -0,0 +1,204 @@
+  // fill-in scalar values and gradients
+  {
+    DoFHandler<dim> dof_scalar(tr);
+    dof_scalar.distribute_dofs(fe_scalar);
+    FEValues<dim> fe_values_scalar (fe_scalar, quadrature,
+                                    update_values | update_gradients);
+    fe_values_scalar.reinit (dof_scalar.begin_active());
+
+    for (unsigned int i=0; i<fe_scalar.dofs_per_cell; ++i)
+      {
+        const unsigned int i_node = fe_scalar.system_to_base_index(i).second;
+        Assert (i_node < fe_scalar.dofs_per_cell, ExcInternalError());
+        for (unsigned int q=0; q<quadrature.size(); ++q)
+          {
+            scalar_values[i_node][q] = fe_values_scalar.shape_value(i,q);
+            scalar_gradients[i_node][q] = fe_values_scalar.shape_grad(i,q);
+          }
+      }
+  }
+
+  FEValues<dim> fe_values (fe, quadrature,
+                           update_values | update_gradients | update_quadrature_points);
+  fe_values.reinit (dof.begin_active());
+
+  // let the FEValues object compute the
+  // divergences at quadrature points
+  FEValuesExtractors::Tensor<2> extractor(0);
+  /*
+  std::vector<Tensor<1,dim> > divergences (quadrature.size());
+  fe_values[extractor]
+  .get_function_divergences (fe_function, divergences);
+
+  std::vector<types::global_dof_index> local_dof_indices (fe.dofs_per_cell);
+  dof.begin_active()->get_dof_indices (local_dof_indices);
+  */
+
+  for (unsigned int i=0; i<fe.dofs_per_cell; ++i)
+    {
+      const auto val = fe_values[extractor].value (i,0);
+      // find out which component is non-zero
+      TableIndices<2> nonzero_ind;
+      for (unsigned int k =0; k < Tensor<2,dim>::n_independent_components; ++k)
+        {
+          nonzero_ind = dealii::Tensor<2,dim>::unrolled_to_component_indices(k);
+          if (std::abs(val[nonzero_ind])>1e-10)
+            break;
+        }
+
+      // const unsigned int i_group = fe.system_to_base_index(i).first.first;
+      // // the index of the non-zero shape function
+      // const unsigned int i_index = fe.system_to_base_index(i).first.second;
+      // the support point (node) id
+      const unsigned int i_node = fe.system_to_base_index(i).second;
+
+      deallog << "i=" << i << " ii="<<nonzero_ind[0]<<" jj="<<nonzero_ind[1]<<" node="<<i_node<<std::endl;
+
+      // const TableIndices<2> indices = dealii::Tensor<2,dim>::unrolled_to_component_indices(comp);
+
+      for (unsigned int q=0; q<quadrature.size(); ++q)
+        {
+          const auto val_q = fe_values[extractor].value (i,q);
+          const auto grad_q = fe_values[extractor].gradient(i,q);
+          const auto div_q = fe_values[extractor].divergence (i,q);
+          deallog << "  q_point=" << fe_values.quadrature_point(q) << std::endl
+                  << "    value= " << val_q << std::endl
+                  << "    grad= " << grad_q << std::endl
+                  << "    div= " << div_q << std::endl;
+          // check value and gradients:
```

Comment:
@mac-a next several lines that check values, gradients and divergence of tesnor-valued shape-function is the new tests I added. Should be enough for our purposes. Have a brief look at let me know if you want to change something.